rclcpp/include/rclcpp/strategies/static_memory_strategy.hpp

```diff
@@ -0,0 +1,159 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP_RCLCPP_STATIC_MEMORY_STRATEGY_HPP_
+#define RCLCPP_RCLCPP_STATIC_MEMORY_STRATEGY_HPP_
+
+#include <rclcpp/memory_strategy.hpp>
+
+namespace rclcpp
+{
+
+namespace memory_strategies
+{
+
+namespace static_memory_strategy
+{
+
+class StaticMemoryStrategy : public memory_strategy::MemoryStrategy
+{
+
+public:
+  StaticMemoryStrategy()
+  {
+    memset(_memory_pool, 0, _pool_size);
+    memset(_subscriber_pool, 0, _max_subscribers);
+    memset(_service_pool, 0, _max_services);
+    memset(_client_pool, 0, _max_clients);
+    memset(_guard_condition_pool, 0, _max_guard_conditions);
+    _pool_seq = 0;
+    _exec_seq = 0;
+
+    for (size_t i = 0; i < _pool_size; ++i) {
+      _memory_map[_memory_pool[i]] = 0;
```

Comment:
Use of a map here will use dynamic allocation, since the internal storage is a tree and the nodes are allocated dynamically. There seems to be a way to avoid this, but it isn't trivial:

http://stackoverflow.com/a/13051059/671658

Maybe we should consider a different strategy for this.